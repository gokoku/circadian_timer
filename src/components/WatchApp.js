import React from 'react'
import '../App.css'
import SetTime from '../containers/SetTime'

export default class WatchApp extends React.Component {
  constructor(props) {
    super(props)
    this.displayTime = this.displayTime.bind(this)
  }

  displayTime() {
    const counter = this.props.timerCounter
    const sec = '0' + (counter % 60)
    const min = '0' + Math.floor(counter / 60)
    return (
      <div>
        <span className="min">{min.substr(min.length - 2, 2)}</span>
        <span className="sec">{sec.substr(sec.length - 2, 2)}</span>
      </div>
    )
  }

  render() {
    const {
      timerId,
      moveStartTimer,
      moveStopTimer,
      clearTimer,
      timeOver
    } = this.props
    return (
      <div className={timeOver ? 'App over' : 'App'}>
        <div className="display">{this.displayTime()}</div>
        <div className="button">
          {timerId ? (
            <button
              onClick={() => {
                moveStopTimer()
              }}
            >
              <img
                src="images/icon_pause.svg"
                className="icon_pause"
                alt="icon pause"
              />
            </button>
          ) : (
            <button
              onClick={() => {
                moveStartTimer()
              }}
            >
              <img
                src="images/icon_play.svg"
                className="icon_play"
                alt="icon play"
              />
            </button>
          )}
          <button
            onClick={() => {
              clearTimer()
            }}
          >
            <img
              src="images/icon_clear.svg"
              className="icon_clear"
              alt="icon clear"
            />
          </button>
        </div>
        <SetTime />
      </div>
    )
  }
}
