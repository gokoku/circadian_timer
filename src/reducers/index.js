const initialState = {
  timerCounter: 0,
  timerId: null,
  limitTime: 0,
  timeOver: false
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'START_TIMER':
      return {
        ...state,
        timerId: action.payload.timerId
      }
    case 'STOP_TIMER':
      return {
        ...state,
        timerId: null
      }
    case 'INCREMENT_COUNTER':
      return {
        ...state,
        timerCounter: state.timerCounter + 1
      }
    case 'RESET_TIMER':
      return {
        ...state,
        timerId: null,
        timerCounter: 0,
        timeOver: false
      }
    case 'SET_TIME':
      return {
        ...state,
        limitTime: action.payload.limitTime
      }
    case 'TIME_OVER':
      return {
        ...state,
        timeOver: action.payload.timeOver
      }
    default:
      return state
  }
}
