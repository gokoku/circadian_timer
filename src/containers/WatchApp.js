import { connect } from 'react-redux'
import WatchApp from '../components/WatchApp'
import { moveTimer, clearTimer } from '../actions'

function mapStateToProps({ timeOver, timerCounter, timerId }) {
  return {
    timerCounter,
    timerId,
    timeOver
  }
}

function mapDispatchToProps(dispatch) {
  return {
    moveTimer() {
      dispatch(moveTimer())
    },
    clearTimer() {
      dispatch(clearTimer())
    }
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(WatchApp)
