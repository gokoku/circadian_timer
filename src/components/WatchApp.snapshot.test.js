import React from 'react'
import renderer from 'react-test-renderer'
import WatchApp from './WatchApp'
import { reducer } from '../reducers'
import { createStore } from 'redux'
import { Provider } from 'react-redux'

const store = createStore(reducer)

it('WatchAppコンポーネントのスナップショットテスト', () => {
  const result = renderer
    .create(
      <Provider store={store}>
        <WatchApp />
      </Provider>
    )
    .toJSON()

  expect(result).toMatchSnapshot()
})
