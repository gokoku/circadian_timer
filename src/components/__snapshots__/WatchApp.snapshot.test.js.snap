// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WatchAppコンポーネントのスナップショットテスト 1`] = `
<div
  className="App"
>
  <div
    className="display"
  >
    <div>
      <span
        className="min"
      >
        aN
      </span>
      <span
        className="sec"
      >
        aN
      </span>
    </div>
  </div>
  <div
    className="button"
  >
    <button
      onClick={[Function]}
    >
      <img
        alt="icon play"
        className="icon_play"
        src="images/icon_play.svg"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <img
        alt="icon clear"
        className="icon_clear"
        src="images/icon_clear.svg"
      />
    </button>
  </div>
  <ul>
    <li
      className="active"
      onClick={[Function]}
    >
      0
       min
    </li>
    <li
      onClick={[Function]}
    >
      15
       min
    </li>
    <li
      onClick={[Function]}
    >
      30
       min
    </li>
    <li
      onClick={[Function]}
    >
      45
       min
    </li>
  </ul>
</div>
`;
